### PR TITLE
feature #39 backend:Integrate analysis script with backend API

### PR DESCRIPTION
**Pull request by nancymuyeh**

### CODE REVIEW

**Comment by Bansikah** [comment]
@stephane-segning feature(backend) integrate analysis script with backend API by @nancymuyeh & @Bansikah

File: templates/test.html
Comment (stephane-segning) [inline templates/test.html:1]: What is the purpose of this file for this ticket please?

File: .gitignore
Comment (stephane-segning) [inline .gitignore:1]: What is the purpose of this file for this ticket please?

File: analysis.py
Comment (stephane-segning) [inline analysis.py:1]: What is the purpose of this file for this ticket please?

File: analyze.ipynb
Comment (stephane-segning) [inline analyze.ipynb:1]: What is the purpose of this file for this ticket please?

File: doc/38-implement-data-analysis-script.md
Comment (stephane-segning) [inline doc/38-implement-data-analysis-script.md:1]: What is the purpose of this file for this ticket please?

File: templates/base.html
Comment (stephane-segning) [inline templates/base.html:1]: What is the purpose of this file for this ticket please?

File: templates/results.html
Comment (stephane-segning) [inline templates/results.html:1]: What is the purpose of this file for this ticket please?

File: templates/input.html
Comment (stephane-segning) [inline templates/input.html:1]: Please remove changes that are not yours

File: app.py
Comment (stephane-segning) [inline app.py:1]: This file seems outdated or contain changes that are not yours. Please fix this

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good work guys! But still a problem, at least for me. This ticket is where you bind
- #38 with
- #34 

as explained here #39. So normally you're not done yet, if those are not done. Plus, your code contains changes that are not coming from you. You probably didn't pulled changes from the main branch before adding your changes

**Comment by Bansikah** [comment]
@stephane-segning Removed files(global) removed files not needed

File: .gitignore
Comment (stephane-segning) [inline .gitignore:1]: Cancel the change in this file

File: app.py
Comment (stephane-segning) [reply app.py:1]: This is still not fixed

File: app.py [hunk 1]
```diff
@@ -92,9 +100,94 @@ def save_to_json(data):
     with open(os.path.join(directory, 'scraped_data.json'), 'w', encoding='utf-8') as jsonfile:
         json.dump(data, jsonfile, ensure_ascii=False, indent=2)
 
-@app.route('/api/start-analysis', methods = ['GET', 'POST'])
-def analysis():
-	return render_template('test.html')
+
+#Developing an api end point that will trigger the analysis script
+class AnalyzeData(Resource):
```
Comment (stephane-segning) [inline app.py:105]: I suppose this is your main content? Put it in another file please

File: requirements.txt
Comment (stephane-segning) [inline requirements.txt:1]: Is is really working? Because https://pypi.org/project/chart.jsflask-bootstrap does not exist

File: templates/base.html
Comment (stephane-segning) [reply templates/base.html:1]: You didn't cancelled this

File: templates/input.html
Comment (stephane-segning) [reply templates/input.html:1]: You didn't cancelled this

File: templates/results.html
Comment (stephane-segning) [reply templates/results.html:1]: You didn't cancelled this

File: templates/test.html
Comment (stephane-segning) [reply templates/test.html:1]: You didn't cancelled this

**Review by stephane-segning — CHANGES_REQUESTED** [review]
You didn't reviewed my points. Please, cancel changes that are not related to your ticket.

**Comment by Bansikah** [comment]
@stephane-segning [feature(backend) integrating analysis script with backend api correction
There is a file in the branch test.ipynb please you it to test the code, i will delete after the code has been reviewed

**Comment by Bansikah** [comment]
@stephane-segning  resolved merge conflicts

**Review by stephane-segning — APPROVED** [review]
Perfect! This is an example of good code! Well done @nancymuyeh @Bansikah

**Comment by Bansikah** [comment]
Thank you @stephane-segning

**Pull request merged**